packages/lib/smart-account/src/lukso/LuksoIndexer.ts

```diff
@@ -0,0 +1,60 @@
+import { ApolloClient, InMemoryCache, gql } from '@apollo/client';
+import { getAddress } from 'ethers/lib/utils';
+
+export class LuksoIndexer {
+    private client: ApolloClient<any>;
+
+    constructor() {
+        //official lukso indexer url
+        const indexerUrl = 'https://envio.mainnet.lukso.dev/v1/graphql';
+        const cache = new InMemoryCache();
+
+        //Create new ApolloClient instance
+        const client = new ApolloClient({
+            // Provide required constructor fields
+            cache: cache,
+            uri: indexerUrl,
+        });
+
+        this.client = client;
+    }
+
+    public async resolveAddress(address: string): Promise<string | undefined> {
+        const query = gql`
+            query MyQuery {
+                Profile(where: { id: { _ilike: "${address}" } }) {
+                    fullName
+                }
+            }
```

Comment:
i think that gql queries shall be built more like this:

```ts
            query getProfile($address: string!) {
                Profile(where: { id: { _ilike: $address } }) {
                    fullName
                }
            }
```
also not sure about the API here, but there is probably also `_eq` to get a quicker result?

Reply:
maybe we could also use gql codegen to streamline the process of generating gql functions.  
Might also get us fully E2E typescript support
https://github.com/dotansimha/graphql-code-generator

Reply:
I changed the query name to a more expressive name. I think the reason for not using _eq is to support checksum addresses as well.